pkg/e2e/setup.go

```diff
@@ -99,44 +101,52 @@ func setupCluster() (err error) {
 		return useKubeconfig()
 	}
 
+	provisioner, err := provisioners.ClusterProvisioner()
+
+	if err != nil {
+		return fmt.Errorf("error getting cluster provisioning client: %v", err)
+	}
+
 	// create a new cluster if no ID is specified
 	if state.Cluster.ID == "" {
 		if state.Cluster.Name == "" {
 			state.Cluster.Name = clusterName()
 		}
 
-		if state.Cluster.ID, err = OSD.LaunchCluster(); err != nil {
+		if state.Cluster.ID, err = provisioner.LaunchCluster(); err != nil {
 			return fmt.Errorf("could not launch cluster: %v", err)
 		}
 	} else {
 		log.Printf("CLUSTER_ID of '%s' was provided, skipping cluster creation and using it instead", state.Cluster.ID)
 
-		cluster, err := OSD.GetCluster(state.Cluster.ID)
+		cluster, err := provisioner.GetCluster(state.Cluster.ID)
 		if err != nil {
 			return fmt.Errorf("could not retrieve cluster information from OCM: %v", err)
 		}
 
 		state.Cluster.Name = cluster.Name()
 		log.Printf("CLUSTER_NAME set to %s from OCM.", state.Cluster.Name)
 
-		state.Cluster.Version = cluster.Version().ID()
+		state.Cluster.Version = cluster.Version()
 		log.Printf("CLUSTER_VERSION set to %s from OCM.", state.Cluster.Version)
 
-		state.CloudProvider.CloudProviderID = cluster.CloudProvider().ID()
+		state.CloudProvider.CloudProviderID = cluster.CloudProvider()
 		log.Printf("CLOUD_PROVIDER_ID set to %s from OCM.", state.CloudProvider.CloudProviderID)
 
-		state.CloudProvider.Region = cluster.Region().ID()
+		state.CloudProvider.Region = cluster.Region()
 		log.Printf("CLOUD_PROVIDER_REGION set to %s from OCM.", state.CloudProvider.Region)
+
+		log.Printf("Found addons: %s", strings.Join(cluster.Addons(), ","))
 	}
 
 	metadata.Instance.SetClusterName(state.Cluster.Name)
 	metadata.Instance.SetClusterID(state.Cluster.ID)
 
-	if err = OSD.WaitForClusterReady(); err != nil {
+	if err = cluster.WaitForClusterReady(provisioner, state.Cluster.ID); err != nil {
```

Comment:
This may sound weird but having a `cluster` method that we pass a provisioner to, feels somewhat clunky, especially, then, using a `provisioner` method to the grab the Kubeconfig

Let me know if that makes sense, it's surprisingly hard for me to articulate right now lol

Reply:
Yeah, good point -- this should probably just be a method that has the kubeconfig. +1. I'll change this.

This has the added benefit of *not* making an OCM call every time, too.

Reply:
Ugh, I forgot this also waits for the provisioner to report "ready." Let me think on this.